src/main/scala/io/iohk/ethereum/vm/ProgramContext.scala

```diff
@@ -1,13 +1,51 @@
 package io.iohk.ethereum.vm
 
-import akka.util.ByteString
+import io.iohk.ethereum.domain._
 
+
+object ProgramContext {
+  def apply(stx: SignedTransaction, blockHeader: BlockHeader, world: WorldStateProxy): ProgramContext = {
+    import stx.tx
+
+    val senderAddress = stx.recoveredSenderAddress.get // FIXME: get, it should be validated but...
```

Comment:
> What is the use case for an invalid SignedTransaction?

I have no idea. @adamsmo ?

> Do we still need the Transaction / SignedTransaction separation?

Yes, we need to be able to create a `Transaction` without a signature.